### DISCUSSION TITLE
I have a question about the documentation of "How do i use a plugin not packaged in nixpkgs?"

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2083] [Q&A] [answered]
About this: https://nix-community.github.io/nixvim/user-guide/faq.html?highlight=plugins#how-do-i-use-a-plugin-not-packaged-in-nixpkgs.
I understand how to do it, and I tried it, and I succeeded.
But, this means the particular plugin that is not packaged in nixvim or nixpkgs, will always be the same version due to the `rev` and/or `hash` value. Even if I do `sudo nixos-rebuild switch --upgrade`, the plugin will for ever be the same version.
Do I understand it correctly?
If yes, how can I auto update it?

Thanks in advance.

**Comment by MattSturgeon** [comment, marked as answer]
Yes, this is the same as how nixpkgs is implemented. Contributors manually update versions and hashes in PRs to nixpkgs.

However, if you're using flakes, you could take advantage of your flake inputs (and therefore your flake lock) to handle this for you when you run `nix flake update`.

I don't have a solid example ready and I don't fancy typing one up on my phone keyboard, but essentially you would have to pass your flake inputs into your modules via `specialArgs`, then use the input in question as the `src` attr.

Alternatively, you could define the package elsewhere in your flake (e.g. your flakes `packages` output) and then pass that into your modules somehow.

But there are limitations to this technique; you'll generally need to follow a git branch with flake inputs (although inputs don't need to be other flakes). This means it's hard to follow a stable version of some software; you could pin to a release tag, but then you won't get updates since release tags generally don't change.

**Reply by refaelsh** [reply]
U understand. Thank you for the answer.

> Yes, this is the same as how nixpkgs is implemented. Contributors manually update versions and hashes in PRs to nixpkgs.

Oh yes, I get it now.

> I don't fancy typing one up on my phone keyboard

Please don't :-) We all love you :-)
